sway/commands/output/non_desktop.c

```diff
@@ -0,0 +1,106 @@
+#include <strings.h>
+#include <wlr/types/wlr_output_damage.h>
+#include "sway/commands.h"
+#include "sway/config.h"
+#include "sway/output.h"
+#include "sway/server.h"
+#include "log.h"
+#include "util.h"
+
+static int find_non_desktop_output(const void *item, const void *data)
+{
+	const struct sway_output_non_desktop *output = item;
+	const char *name = data;
+	return strcmp(output->wlr_output->name, name);
+}
+
+static int find_output(const void *item, const void *data)
+{
+	const struct sway_output *output = item;
+	const char *name = data;
+	return strcmp(output->wlr_output->name, name);
+}
+
+struct cmd_results *output_cmd_non_desktop(int argc, char **argv) {
+	if (argc == 0) {
+		return cmd_results_new(CMD_INVALID, "Missing non_desktop argument");
+	}
+
+	struct output_config *oc = config->handler_context.output_config;
+
+	if (strcmp(oc->name, "*") == 0) {
+		return cmd_results_new(CMD_INVALID,
+			"Cannot apply non_desktop to all outputs");
+	}
+
+	if (parse_boolean(argv[0], true)) {
+		int index = list_seq_find(root->outputs, find_output, oc->name);
```

Comment:
We already have a function to do this:

```
struct sway_output *sway_output = all_output_by_name_or_id(oc->name);
```

It also handles output identifiers.

Reply:
`all_output_by_name_or_id` searches in `root->all_outputs`,  is it better? I didnt get the difference between the two lists.